internal/envoy/envoy.go

```diff
@@ -309,13 +309,20 @@ func (srv *Server) addTraceConfig(traceOpts *config.TracingOptions, bootCfg *env
 	return nil
 }
 
-func (srv *Server) handleLogs(stdout io.ReadCloser) {
+func (srv *Server) handleLogs(rc io.ReadCloser) {
+	defer rc.Close()
+
 	logFormatRE := regexp.MustCompile(`^[[]LOG_FORMAT[]](.*?)--(.*?)--(.*?)$`)
 	fileNameAndNumberRE := regexp.MustCompile(`^(\[[a-zA-Z0-9/-_.]+:[0-9]+])\s(.*)$`)
 
-	s := bufio.NewScanner(stdout)
-	for s.Scan() {
-		ln := s.Text()
+	s := bufio.NewReader(rc)
+	for {
+		ln, err := s.ReadString('\n')
+		if err != nil {
+			log.Error().Err(err).Msg("failed to read log")
+			continue
```

Comment:
I think the only error we should break on would be `io.EOF`:

```go
if err == io.EOF {
    break
} else if err != nil {
   //...
}
```

Reply:
also we might need exponential backoff on other types of errors? Not sure the best way to handle that

Reply:
👍 Fixed

Reply:
@calebdoxsey I think we don't need, it's hard to think any error in this case other than io.EOF.